config/initializers/puppet.rb

```diff
@@ -12,7 +12,9 @@
     :priority => 100
 end
 
-FactImporter.register_fact_importer :puppet, PuppetFactImporter, true
+Foreman::Plugin.register :foreman_puppet do
+  fact_registry.register(:puppet, PuppetFactImporter, true)
+end
```

Comment:
IIUC, the changes to `config/initializers/assets.rb` and `config/environments/test.rb` are are due to this - since you are registering a plugin that isn't really a plugin to get to the registry. 
You can instead access the registry directly with:
```suggestion
Foreman::Plugin.fact_registry.register(:puppet, PuppetFactImporter, true)
```
Or take an approach like other registries (e.g. the report origin registry), of hard coding puppet in the default values.